frontend/src/routers/RoadmapRouter.tsx

```diff
@@ -91,9 +99,42 @@ const RoadmapRouterComponent = ({ userInfo }: { userInfo: UserInfo }) => {
 
   useEffect(() => {
     // Try to select roadmap given in route parameters
-    if (roadmapId) {
-      dispatch(roadmapsActions.selectCurrentRoadmap(+roadmapId));
+    if (urlRoadmapId) {
+      dispatch(roadmapsActions.selectCurrentRoadmap(+urlRoadmapId));
     }
+  }, [dispatch, urlRoadmapId]);
+
+  useEffect(() => {
+    // If selectedRoadmapId changes, update the url to match that
+    // (for cases where another browser tab switches roadmapId and the state gets synced into this tab)
+    if (previousRoadmapId && selectedRoadmapId !== previousRoadmapId) {
+      let newPath: string;
+      const roadmapIdRegex = new RegExp('roadmap/\\d+/');
```

Comment:
```suggestion
      const roadmapIdRegex = new RegExp('^/roadmap/\\d+/');
```

The regex could be anchored, even though we are unlikely to add other routes that would match

Reply:
The path has a leading `/`